### DISCUSSION TITLE
Delete repeater item without need to save the form

### DISCUSSION THREAD

**Discussion by Vs-tev** [filamentphp/filament#8922] [Help]
Hello, I'm wondering if there is a way to delete repeater items (or even sort items) and save the changes directly afterward without saving the entire form. I know there is a way to customize repeater action objects with  `deleteAction(),` and `reorderAction()` which give you certain options, but until now, I haven't really found a way to do this. Does anyone know or has anyone already done something like this? 

Thanks in advance

**Comment by kombiahrk** [comment]
->deleteAction(
                                    function (Action $action) {
                                        $action->requiresConfirmation();
                                        $action->before(function ($state, array $arguments) {
                                            //dd($arguments['item']);
                                            //dd($state['record-41']['id']);
                                            $key = $arguments['item'];
                                            $record = $state[$key];
                                            if (isset($record['id'])) {
                                                $item = OrderItem::find($record['id']);
                                                //dd($item);
                                                if ($item) {
                                                    // Calculate the stock reduction
                                                    $quantityReduction = $item->quantity;
                                                    $unit = $item->unit_id;
                                                    //dd($quantityReduction);
                                                    // Retrieve the associated product
                                                    $product = Product::find($item->product_id);
                                                    //dd($product);
                                                    if ($product) {
                                                        if ($unit == $product->unit_id) {
                                                            $product->quantity -= $quantityReduction;
                                                        } elseif ($unit == $product->alt_unit_id) {
                                                            $product->quantity -= $quantityReduction / $product->alt_unit_value;
                                                        }
                                                        // Reduce the stock in the product
                                                        $product->save();
                                                    }
                                                    $item->forceDelete();
                                                    Notification::make()
                                                        ->title('Saved')
                                                        ->success()
                                                        ->send();
                                                }
                                            }
                                        });
                                    }
                                )
Above code may helps you

**Comment by iammuttaqi** [comment]
```php
->deleteAction(function (Action $action) {
  $action->requiresConfirmation();
  $action->before(function ($state, array $arguments) {
      //dd($arguments['item']);
      //dd($state['record-41']['id']);
      $key = $arguments['item'];
      $record = $state[$key];
      if (isset($record['id'])) {
          $item = OrderItem::find($record['id']);
          //dd($item);
          if ($item) {
              // Calculate the stock reduction
              $quantityReduction = $item->quantity;
              $unit = $item->unit_id;
              //dd($quantityReduction);
              // Retrieve the associated product
              $product = Product::find($item->product_id);
              //dd($product);
              if ($product) {
                  if ($unit == $product->unit_id) {
                      $product->quantity -= $quantityReduction;
                  } elseif ($unit == $product->alt_unit_id) {
                      $product->quantity -= $quantityReduction / $product->alt_unit_value;
                  }
                  // Reduce the stock in the product
                  $product->save();
              }
              $item->forceDelete();
              Notification::make()
                  ->title('Saved')
                  ->success()
                  ->send();
          }
      }
  });
});
```

Same this. leaving it here to just to have a nice markdown view ✌😅
